### PR TITLE
Event Pages are Preview-able

### PR DESCRIPTION
**Pull request by gescobardev**
## Description
You can preview a location's event page, for example `pittsburgh-health-center/events`
Currently, if an event is featured, the text above the event will say - "In the spotlight at" but not display the name of the location. So, now it should show for example- `In the spotlight at VA Pittsburgh health care`

In order to display the events data, some functionality needed to be made into `liquid.filters`. 
I added unit tests to each of those functions.

## Testing done
Unit tests have been written for filter functions

## Screenshots
Events page can now be seen in preview along with upcoming/featured event data
Screen Shot 2021-06-08 at 5 28 11 PM

Screen Shot 2021-06-08 at 5 34 48 PM

## Acceptance criteria
- [ ] All tests pass
- [ ] Can view on preview server

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/filters/liquid.js [hunk 1]
```diff
@@ -755,4 +756,130 @@ module.exports = function registerFilters() {
...
+  liquid.filters.paginatePages = (page, items, aria) => {
+    const perPage = 10;
+
+    let ariaLabel = aria;
+
+    if (typeof ariaLabel === 'undefined') {
+      ariaLabel = '';
+    } else {
+      ariaLabel = ` of ${ariaLabel}`;
+    }
```
Comment (mpelzsherman) [inline src/site/filters/liquid.js:807]:
```suggestion
    const ariaLabel = aria ? ` of ${aria}` : '';
```

File: src/site/filters/liquid.js [hunk 2]
```diff
@@ -755,4 +756,130 @@ module.exports = function registerFilters() {
...
+          } else {
+            length = numPageLinks;
+
+            if (pageNum + numPageLinks > pagedEntities.length) {
+              start = pagedEntities.length - numPageLinks;
+            } else {
+              start = pageNum;
+            }
+          }
+          for (let num = start; num < start + length; num++) {
```
Comment (mpelzsherman) [inline src/site/filters/liquid.js:848]: Please add a blank line above this line.

File: src/site/filters/liquid.unit.spec.js [hunk 3]
```diff
@@ -4,9 +4,181 @@ import { expect, assert } from 'chai';
 import registerFilters from './liquid';
 import vetCenterData from '../layouts/tests/vet_center/fixtures/vet_center_escanaba_data';
 import featuredContentData from '../layouts/tests/vet_center/fixtures/featuredContentData.json';
+import eventListingMockData from '../layouts/tests/vamc/fixtures/eventListingMockData.json';
+// import event_listing_mock_data2 from '../layouts/tests/vamc/fixtures/event_listing_mock_data2.json';
```
Comment (mpelzsherman) [inline src/site/filters/liquid.unit.spec.js:8]: Please remove comment

File: src/site/filters/liquid.unit.spec.js [hunk 4]
```diff
@@ -4,9 +4,181 @@ import { expect, assert } from 'chai';
...
+describe('paginatePages', () => {
+  it('passing in less than 10 events', () => {
+    const result = liquid.filters.paginatePages(
+      eventListingMockData,
+      eventListingMockData.reverseFieldListingNode.entities,
+    );
+
+    expect(result.pagedItems.length).to.be.below(11);
+    expect(result.paginator.next).to.be.null;
+  });
```
Comment (mpelzsherman) [inline src/site/filters/liquid.unit.spec.js:179]: Can we add a test for > 10?

File: src/site/filters/liquid.js (hunk 1 above)
Comment (gescobardev) [reply src/site/filters/liquid.js:807]: this is even better. will do

File: src/site/filters/liquid.unit.spec.js (hunk 4 above)
Comment (gescobardev) [reply src/site/filters/liquid.unit.spec.js:179]: Yes we can! I was in the process of adding. I made a separate json for it which is why I commented out above. 😁

**Comment by gescobardev** [comment]
@mpelzsherman Thank you, changes have been made

File: src/site/filters/liquid.unit.spec.js [hunk 5]
```diff
@@ -4,9 +4,194 @@ import { expect, assert } from 'chai';
 import registerFilters from './liquid';
 import vetCenterData from '../layouts/tests/vet_center/fixtures/vet_center_escanaba_data';
 import featuredContentData from '../layouts/tests/vet_center/fixtures/featuredContentData.json';
+import eventListingMockData from '../layouts/tests/vamc/fixtures/eventListingMockData.json';
+import eventListingMockData2 from '../layouts/tests/vamc/fixtures/eventListingMockData2.json';
```
Comment (mpelzsherman) [inline src/site/filters/liquid.unit.spec.js:8]: Do we really need both fixtures, or could we just use `eventListingMockData2` and use [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) to reduce the size?

File: src/site/layouts/vet_center.drupal.liquid [hunk 6]
```diff
@@ -19,7 +19,7 @@
                         </div>
                     {% endif %}
 
-                    <nav id="table-of-contents">
+                    <nav id="table-of-contents" aria-labelledby="on-this-page">
```
Comment (mpelzsherman) [inline src/site/layouts/vet_center.drupal.liquid:22]: Did you make this change? I think this may be the result of rebase or merge issues.

Comment (gescobardev) [reply src/site/layouts/vet_center.drupal.liquid:22]: No, this is not part of my changes.

File: src/site/filters/liquid.unit.spec.js (hunk 5 above)
Comment (gescobardev) [reply src/site/filters/liquid.unit.spec.js]: Yes, that makes more sense 👌

**Review by mpelzsherman — APPROVED** [review]

**Pull request merged**